pipeline-runner/R/gem2s-1-download_cellranger_files.R

```diff
@@ -1,29 +1,36 @@
-download_cellranger <- function(input, pipeline_config) {
+#' Download cellranger files from S3
+#'
+#' @param input The input object from the request
+#' @param pipeline_config result of \code{load_config}
+#' @param prev_out \code{list()} that is added to in each with gem2s task.
+#'
+#' @return list where 'output' slot has config used by \code{load_cellranger}
+#' @export
+#'
+download_cellranger_files <- function(input, pipeline_config, prev_out = list()) {
   project_id <- input$projectId
   sample_names <- input$sampleNames
   sample_uuids <- input$sampleIds
 
-  print_config(1, "Cellranger", input, pipeline_config, list())
-
   s3 <- paws::s3(config = pipeline_config$aws_config)
 
-  fnames <- c("features.tsv.gz", "barcodes.tsv.gz", "matrix.mtx.gz")
+  cellranger_fnames <- c("features.tsv.gz", "barcodes.tsv.gz", "matrix.mtx.gz")
   unlink("/input", recursive = TRUE)
 
   for (sample in sample_uuids) {
-    for (fname in fnames) {
-      gem_key <- file.path(project_id, sample, fname)
+    message("\nSample --> ", sample)
+
+    for (cellranger_fname in cellranger_fnames) {
+      gem_key <- file.path(project_id, sample, cellranger_fname)
```

Comment:
Is it possible that one of the file names in `cellranger_fnames` will not be in `file.path`? What will happen in that case?

Reply:
the `file.path` is used to construct the S3 object location that we are going to download. If they are not there then the upload failed earlier on but gem2s was allowed to start anyways. If that happened then an error will be thrown when it tries to get the S3 object from S3. I 'm not sure that checking that the S3 objects are there and throwing an error if they are not provides additional benefit. If they are not there, GEM2S can't run and we have to throw an error anyways. It would just add code/time to list the S3 objects first with the same endpoint - throwing an error.

Reply:
I just checked and if the object isn't in S3 then the error is thrown right away (I changed the gem key to download blah.tsv.gz):

![image](https://user-images.githubusercontent.com/15719520/129783645-a80f4861-5c4e-4cef-b243-8a724c2a78b9.png)